### PR TITLE
[network checks] 

### PR DESCRIPTION
**Pull request by remh**
### What does this PR do?

Use a global counter for the pool size.
This way our assertion on the max number of threads will be accurate.

### Motivation

See #3080 for details

### CODE REVIEW

File: checks/network_checks.py [hunk 1]
```diff
@@ -117,7 +128,7 @@ def restart_pool(self):
     def check(self, instance):
         if not self.pool_started:
             self.start_pool()
-        if threading.activeCount() > 5 * self.pool_size + 5: # On Windows the agent runs on multiple threads so we need to have an offset of 5 in case the pool_size is 1
+        if threading.activeCount() > 5 * _global_current_pool_size + 1:
```
Comment (ross) [inline checks/network_checks.py]: Does this need a `global _global_current_pool_size`? It appears in the other 2 places this is used.

File: checks/network_checks.py [hunk 2]
```diff
@@ -18,6 +18,8 @@
 MAX_LOOP_ITERATIONS = 1000
 FAILURE = "FAILURE"
 
+_global_current_pool_size = 0
```
Comment (ross) [inline checks/network_checks.py]: Could this just be a property on the `NetworkCheck` class? Should save the need for the `globals` and tie it to a logically connected class.

File: checks/network_checks.py (hunk 1 above)
Comment (remh) [inline checks/network_checks.py]: It doesn't, in python the need for the global keyword is not needed when you read the value of a variable.

File: checks/network_checks.py [hunk 3]
```diff
@@ -117,7 +125,7 @@ def restart_pool(self):
     def check(self, instance):
         if not self.pool_started:
             self.start_pool()
-        if threading.activeCount() > 5 * self.pool_size + 5: # On Windows the agent runs on multiple threads so we need to have an offset of 5 in case the pool_size is 1
+        if threading.activeCount() > 5 * NetworkCheck._global_current_pool_size + 1:
```
Comment (olivielpeau) [inline checks/network_checks.py]: why change the offset from `5` to `1`?

Comment (remh) [inline checks/network_checks.py]: The reason was that the windows agent is not using threads to run its different components (collector dogstatsd forwarder) anymore but multiprocessing, which was the reason we had this offset in the first place. However Windows checks use wmi which is itself multithreaded so you're right we should keep this offset. Fixing that.

**Pull request merged**